urbs.py

```diff
@@ -786,6 +907,43 @@ def def_costs_rule(m, cost_type):
             for tm in m.tm for c in m.com_tuples
             if c[1] in m.com_stock)
 
+    elif cost_type == 'Revenue':
+        revenue = 0
+        sell_tuples = commodity_subset_list(m.com_tuples, m.com_sell)
+        for c in sell_tuples:
+            # check com price fix or timeseries
+            if not isinstance(m.commodity.loc[c]['price'], float):
```

Comment:
The "heavy lifting" should not happen in the cost function. Its structure, because it is already long, must stay as simple as possible.